db/migrate/20211207140550_create_user_permission.rb

```diff
@@ -0,0 +1,10 @@
+class CreateUserPermission < ActiveRecord::Migration[6.1]
+  def change
+    create_table :user_permission do |t|
+      t.references :user, null: false, foreign_key: true, index: { unique: true }
+      t.references :provider, null: false, foreign_key: true, index: { unique: true }
```

Comment:
https://apidock.com/rails/v2.2.1/ActiveRecord/ConnectionAdapters/SchemaStatements/add_index

Something like this please


```
 add_index(:user_permission, [:user_id, :provider_id], :unique => true)
```

This will prevent duplications.

An unique index for `both` column as a combination.